packages/material-ui-lab/src/LocalizationProvider/LocalizationProvider.tsx

```diff
@@ -26,17 +38,30 @@ export interface LocalizationProviderProps {
 /**
  * @ignore - do not document.
  */
-const LocalizationProvider: React.FC<LocalizationProviderProps> = (props) => {
+function LocalizationProvider(props: LocalizationProviderProps) {
   const { children, dateAdapter: Utils, dateFormats, dateLibInstance, locale } = props;
   const utils = React.useMemo(
     () => new Utils({ locale, formats: dateFormats, instance: dateLibInstance }),
     [Utils, locale, dateFormats, dateLibInstance],
   );
 
+  const defaultDates: MuiPickersAdapterContextValue<unknown>['defaultDates'] = React.useMemo(() => {
+    return {
+      minDate: utils.date('1900-01-01T00:00:00.000'),
+      maxDate: utils.date('2099-12-31T00:00:00.000'),
+    };
+  }, [utils]);
+
+  const contextValue: MuiPickersAdapterContextValue<unknown> = React.useMemo(() => {
+    return { utils, defaultDates };
+  }, [defaultDates, utils]);
+
   return (
-    <MuiPickersAdapterContext.Provider value={utils}>{children}</MuiPickersAdapterContext.Provider>
+    <MuiPickersAdapterContext.Provider value={contextValue}>
```

Comment:
Since `defaultDates` depends on `utils` we can merge the contexts. `defaultDates` or `utils` cannot change independently.